CareKit/CareKit/Shared/Contact/DetailedContactView.swift

```diff
@@ -0,0 +1,138 @@
+        self.contactView = contactView
+    }
+    
+    public init(contactID: String, storeManager: OCKSynchronizedStoreManager, content: @escaping (_ controller: OCKDetailedContactController) -> CareKitUI.DetailedContactView<Header, Footer>) {
+        contactView = .init(controller: .init(storeManager: storeManager), query: .contactID(contactID), content: content)
+    }
+    
+    public init(contact: OCKAnyContact, contactQuery: OCKContactQuery, storeManager: OCKSynchronizedStoreManager,
+                content: @escaping (_ controller: OCKDetailedContactController) -> CareKitUI.DetailedContactView<Header, Footer>) {
+        contactView = .init(controller: .init(storeManager: storeManager), query: .init(.contact(contact)), content: content)
+    }
+    
+    public init(controller: OCKDetailedContactController, content: @escaping (_ controller: OCKDetailedContactController) -> CareKitUI.DetailedContactView<Header, Footer>) {
+        contactView = .init(controller: controller, content: content)
+    }
+    
+    public func onError(_ perform: @escaping (Error) -> Void) -> DetailedContactView<Header, Footer> {
+        .init(contactView: .init(copying: contactView, settingErrorHandler: perform))
+    }
+}
+
+@available(iOS 14.0, watchOS 7.0, *)
+public extension DetailedContactView where Header == _DetailedContactViewHeader, Footer == _DetailedContactViewFooter {
+    init(contactID: String, storeManager: OCKSynchronizedStoreManager) {
+        self.init(contactID: contactID, storeManager: storeManager) {
+            .init(viewModel: $0.viewModel)
+        }
+    }
+    
+    init(contact: OCKAnyContact,contactQuery: OCKContactQuery, storeManager: OCKSynchronizedStoreManager) {
+        self.init(contact: contact, contactQuery: contactQuery, storeManager: storeManager) {
+            .init(viewModel: $0.viewModel)
+        }
+    }
+    
+    init(controller: OCKDetailedContactController) {
+        contactView = .init(controller: controller) {
+            .init(viewModel: $0.viewModel)
+        }
+    }
+}
+
+// CODE REVIEW: Let's add actions - they are nil right now, see UIKit version
+private extension CareKitUI.DetailedContactView where Header == _DetailedContactViewHeader, Footer == _DetailedContactViewFooter {
+    init(viewModel: ContactViewModel?) {
+        self.init(title: Text(viewModel?.title ?? ""), detail: Text(viewModel?.detail ?? ""), instructions: Text(viewModel?.instructions ?? ""), image: Image(systemName: "person.crop.circle"), disclosureImage: nil, callButton: ContactButton(title: Text("Call"), image: Image(systemName: "phone"), action: nil), messageButton: ContactButton(title: Text("Message"), image: Image(systemName: "text.bubble"), action: nil), emailButton:  ContactButton(title: Text("E-mail"), image: Image(systemName: "envelope"), action: nil), addressButton: AddressButton(title: Text("Address"), detail: Text(viewModel?.address ?? ""), image: Image(systemName: "location"), action: nil))
```

Comment:
This line is really long. Consider putting in line breaks after each argument.

Reply:
Done